File: rest-service/src/main/java/mb/lib/query/db/InsertBlastTargets.java
package mb.lib.query.db;

import java.sql.Connection;

import mb.lib.model.HashID;
import mb.lib.query.model.JobTarget;

public class InsertBlastTargets
{
  private static final String Query = """
    INSERT INTO
      userlogins5.multiblast_job_to_targets (job_digest, organism, target_file)
    VALUES
      (?, ?, ?)
    """;

  private final Connection  con;
  private final HashID      jobID;
  private final JobTarget[] dbs;

  public InsertBlastTargets(
    Connection con,
    HashID jobID,
    JobTarget[] dbs
  ) {
    this.con      = con;
    this.jobID    = jobID;
    this.dbs      = dbs;
  }

  public void run() throws Exception {
    try (var ps = con.prepareStatement(Query)) {
      ps.setBytes(1, jobID.bytes());

      for (var db : dbs) {
        ps.setString(2, db.getOrganism());
        ps.setString(3, db.getTarget());
        ps.addBatch();
      }

      ps.executeBatch();
    }
  }
}

File: rest-service/src/main/kotlin/mb/lib/query/db/InsertBlastTargets.kt
package mb.lib.query.db

import mb.lib.query.model.JobTarget
import org.veupathdb.lib.hash_id.HashID
import java.sql.Connection

@Suppress("ArrayInDataClass")
data class InsertBlastTargets(
  private val con: Connection,
  private val jobID: HashID,
  private val dbs: Array<out JobTarget>,
) {
  companion object {
    // language=postgresql
    private const val Query = """
    INSERT INTO
      multiblast.multiblast_job_to_targets (job_digest, organism, target_file)
    VALUES
      (?, ?, ?)
    """
  }

  fun run() {
    con.prepareStatement(Query)!!.use {
      it.setBytes(1, jobID.bytes)

      for (db in dbs) {
        it.setString(2, db.organism)
        it.setString(3, db.target)
        it.addBatch()
      }

      it.executeBatch()
    }
  }
}
